sentry_streams/src/transformer.rs

```diff
@@ -66,13 +86,142 @@ mod tests {
+    use chrono::Utc;
     use pyo3::ffi::c_str;
     use pyo3::IntoPyObjectExt;
+    use sentry_arroyo::processing::strategies::InvalidMessage;
     use sentry_arroyo::processing::strategies::ProcessingStrategy;
+    use sentry_arroyo::types::Partition;
+    use sentry_arroyo::types::Topic;
     use std::collections::BTreeMap;
     use std::ops::Deref;
     use std::sync::{Arc, Mutex};
 
+    #[test]
+    #[should_panic(
+        expected = "Got exception while processing AnyMessage, Arroyo cannot handle error on AnyMessage"
+    )]
+    fn test_build_map_crashes_on_any_msg() {
+        pyo3::prepare_freethreaded_python();
+        traced_with_gil("test_build_map", |py| {
+            py.run(
+                c_str!("from sentry_streams.pipeline.exception import MessageInvalidError"),
+                None,
+                None,
+            )
+            .expect("Unable to import");
+            let callable = make_lambda(
+                py,
+                c_str!("lambda x: (_ for _ in ()).throw(MessageInvalidError())"),
+            );
+            let submitted_messages = Arc::new(Mutex::new(Vec::new()));
+            let submitted_watermarks = Arc::new(Mutex::new(Vec::new()));
+            let next_step = FakeStrategy::new(submitted_messages, submitted_watermarks, false);
+
+            let mut strategy = build_map(
+                &Route::new("source1".to_string(), vec!["waypoint1".to_string()]),
+                callable,
+                Box::new(next_step),
+            );
+
+            let message = Message::new_any_message(
+                build_routed_value(
+                    py,
+                    "test_message".into_py_any(py).unwrap(),
+                    "source1",
+                    vec!["waypoint1".to_string()],
+                ),
+                BTreeMap::new(),
+            );
+            let _ = strategy.submit(message);
+        });
+    }
+
+    #[test]
+    #[should_panic(
+        expected = "Python map function raised exception that is not sentry_streams.pipeline.exception.MessageInvalidError"
+    )]
+    fn test_build_map_crashes_on_normal_exceptions() {
+        pyo3::prepare_freethreaded_python();
+        traced_with_gil("test_build_map", |py| {
+            py.run(
+                c_str!("from sentry_streams.pipeline.exception import MessageInvalidError"),
+                None,
+                None,
+            )
+            .expect("Unable to import");
+            let callable = make_lambda(py, c_str!("lambda x: {}[0]"));
+            let submitted_messages = Arc::new(Mutex::new(Vec::new()));
+            let submitted_watermarks = Arc::new(Mutex::new(Vec::new()));
+            let next_step = FakeStrategy::new(submitted_messages, submitted_watermarks, false);
+
+            let mut strategy = build_map(
+                &Route::new("source1".to_string(), vec!["waypoint1".to_string()]),
+                callable,
+                Box::new(next_step),
+            );
```

Comment:
There is a lot of copy paste in these three tests. 
Please let's reduce it as it makes reviewing quite slower than it has to be.

The only difference between the strategy generated in the three tests is the lambda. Consider creating a function in the test module that takes the lambda and returns the strategy.

Also you are not usinf anything in the `next_step` strategy. Consider using the Noop strategy: `use sentry_arroyo::processing::strategies::noop::Noop;`